src/main/java/com/example/producerconsumer/services/ProducersConsumersService.java

```diff
@@ -81,6 +90,8 @@ private void validateParameters(int numberOfProducers, int numberOfConsumers) {
     }
 
     public void run(List<QueueProducer<String>> producers, List<QueueConsumer<String>> consumers) {
+        //czy tutaj nie powinniśmy rozdzielić tworzenia pul wątków i inicjacji executorów od samego wrzucenia consumerów/producentów.
+        // rozmiar pól nie jest podany excplicite tylko wynika z rozmiaru przekazanych producentów i koonsumetów.
```

Comment:
niestety nie rozumiem. Ilość wątków uruchomionych chyba zawsze powinna być równa ilości consumer + producentów ? Chyba ze czegoś nie widzę.

Reply:
liczba consumentów i producentów mogłabybyć zmienna? A nawet gdyby założyć stałą liczbę to uważam za dobre odseparowanie przygotowania puli od relacji do samych consumerów/producerów. Zwracam uwagę jak "couplujemy" metody - od jakich parametrów zależą to ma wpływ na późniejsze testy